mdns.h

```diff
@@ -0,0 +1,1206 @@
+		mdns_record_srv_t srv;
+		memset(&srv, 0, sizeof(mdns_record_srv_t));
+		// Read the priority, weight, port number and the discovery name
+		// SRV record format (http://www.ietf.org/rfc/rfc2782.txt):
+		// 2 bytes network-order unsigned priority
+		// 2 bytes network-order unsigned weight
+		// 2 bytes network-order unsigned port
+		// string: discovery (domain) name, minimum 2 bytes when compressed
+		if ((size >= offset + length) && (length >= 8)) {
+			const uint16_t* recorddata = (const uint16_t*)((const char*)buffer + offset);
+			srv.priority = ntohs(*recorddata++);
+			srv.weight = ntohs(*recorddata++);
+			srv.port = ntohs(*recorddata++);
+			offset += 6;
+			srv.name = mdns_string_extract(buffer, size, &offset, strbuffer, capacity);
+		}
+		return srv;
+	}
+
+	static struct sockaddr_in*
+		mdns_record_parse_a(const void* buffer, size_t size, size_t offset, size_t length,
+			struct sockaddr_in* addr) {
+		memset(addr, 0, sizeof(struct sockaddr_in));
+		addr->sin_family = AF_INET;
+#ifdef __APPLE__
+		addr->sin_len = sizeof(struct sockaddr_in);
+#endif
+		if ((size >= offset + length) && (length == 4))
+			addr->sin_addr.s_addr = *(const uint32_t*)((const char*)buffer + offset);
+		return addr;
+	}
+
+	static struct sockaddr_in6*
+		mdns_record_parse_aaaa(const void* buffer, size_t size, size_t offset, size_t length,
+			struct sockaddr_in6* addr) {
+		memset(addr, 0, sizeof(struct sockaddr_in6));
+		addr->sin6_family = AF_INET6;
+#ifdef __APPLE__
+		addr->sin6_len = sizeof(struct sockaddr_in6);
+#endif
+		if ((size >= offset + length) && (length == 16))
+			addr->sin6_addr = *(const struct in6_addr*)((const char*)buffer + offset);
+		return addr;
+	}
+
+	static size_t
+		mdns_record_parse_txt(const void* buffer, size_t size, size_t offset, size_t length,
+			mdns_record_txt_t* records, size_t capacity) {
+		size_t parsed = 0;
+		const char* strdata;
+		size_t separator, sublength;
+		size_t end = offset + length;
+
+		if (size < end)
+			end = size;
+
+		while ((offset < end) && (parsed < capacity)) {
+			strdata = (const char*)buffer + offset;
+			sublength = *(const unsigned char*)strdata;
+
+			++strdata;
+			offset += sublength + 1;
+
+			separator = 0;
+			for (size_t c = 0; c < sublength; ++c) {
+				// DNS-SD TXT record keys MUST be printable US-ASCII, [0x20, 0x7E]
+				if ((strdata[c] < 0x20) || (strdata[c] > 0x7E))
+					break;
+				if (strdata[c] == '=') {
+					separator = c;
+					break;
+				}
+			}
+
+			if (!separator)
+				continue;
+
+			if (separator < sublength) {
+				records[parsed].key.str = strdata;
+				records[parsed].key.length = separator;
+				records[parsed].value.str = strdata + separator + 1;
+				records[parsed].value.length = sublength - (separator + 1);
+			}
+			else {
+				records[parsed].key.str = strdata;
+				records[parsed].key.length = sublength;
+			}
+
+			++parsed;
+		}
+
+		return parsed;
+	}
+
+#ifdef _WIN32
+#undef strncasecmp
+#endif
+
+#ifdef __cplusplus
+}
+#endif
```

Comment:
empty line is missing at EOF

Reply:
we dont control this file - it's an exact copy (plus the git hash) of upstream

Reply:
An almost exact copy .. there was one or two warnings that didn't compile with /WX MSVC compiler switch ..